piggy_bankers_app/View/Profile/CreateTransactionView.swift

```diff
@@ -8,33 +8,50 @@
 import SwiftUI
 
 struct CreateTransactionView: View {
-    @EnvironmentObject var userManager: UserManager
     @Environment(\.dismiss) var dismiss
+    @ObservedObject var transactionManager = TransactionManager()
+    @State private var newTransaction = Transaction()
     
-    @State private var amount: Double = 0.0
-    @State private var description: String = "This is the default description for sending/requesting money."
-    @State private var kid_name: String = ""
+    private let dummyKids = [100,101,102,103,104,105]
     
     var body: some View {
         Form {
+            Section("Placeholder") {
+                Text("ID placeholder")
+                    .lineLimit(0)
+                HStack {
+                    Text("Household placeholder")
+                    Text("Profile placeholder")
+                }
+            }
+            
+            Section("Details") {
+                Text("Date: \(Date())")
                 
-            TextField("How much?", value: $amount, format: .currency(code: "USD"))
+                TextField("How much?", value: $newTransaction.transaction_amount, format: .currency(code: "USD"))
+                
+                TextField("Send or Request?", text: $newTransaction.transaction_type)
+            }
+            
+            Section {
+                TextField("Memo", text: $newTransaction.transaction_memo)
+            }
             
-            TextField("Description", text: $description, axis: .vertical)
-                .lineLimit(3...)
+            Section {
+                TextField("Description", text: $newTransaction.transaction_description)
+            }
             
-            Picker("Select a child.", selection: $kid_name) {
-                ForEach(testKids, id: \.kid_name) { kid in
-                    Text("\(kid.kid_name)")
-                        .tag(kid.kid_name)
+            Picker("Select a child.", selection: $newTransaction.profile_id) {
+                ForEach(dummyKids, id: \.self) { kid in
+                    Text("\(kid)")
+                        .tag(kid)
                 }
             }
             
             Section {
                 Button {
-                    userManager.fundTransaction(name: kid_name, amount: amount)
-                    print("transaction initiated.")
-                    
+                    transactionManager.createTransaction(transaction: newTransaction)
+                    transactionManager.refreshTransactions()
```

Comment:
I wonder if we should just add the transaction manually to our model and push to our database at the same time, so we don't have to refresh the transactions from the DB every update.
Pro- It could run faster
Con- We'd have to manage the transactions in two places
Con- If the transaction doesn't end up in the DB and it shows up in the app then we have a problem

Reply:
at the code's current state, after creating a Transaction (clicking submit) and navigating back to ProfileView, the recently created Transaction was not showing up. This was a work around. Also, not sure why the publisher on all transactions doesn't notice this. Need to look into that

Reply:
Makes sense. I'm cool leaving it as-is for now and taking another look if we have too much latency